plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/delete/TrinoDeleteFilter.java

```diff
@@ -43,6 +47,39 @@ public TrinoDeleteFilter(FileScanTask task, Schema tableSchema, List<IcebergColu
         this.fileIO = requireNonNull(fileIO, "fileIO is null");
     }
 
+    public CloseableIterable<TrinoRow> filterCached(CloseableIterable<TrinoRow> records)
+    {
+        return filterEqDeletesCached(filterPosDeletesCached(records));
+    }
+
+    public CloseableIterable<TrinoRow> filterPosDeletesCached(CloseableIterable<TrinoRow> records)
```

Comment:
I am having a hard time to understand what is actually being cached here.

Can you add a comment or change the 'xxxCached` method names to better match their purpose?

Reply:
Added some comments. Lemme know what you think.

Reply:
Thank you for taking the time to point out the memoization technique in applying the delete filters.
I'm curious what are the downsides of using the memoization technique that make `org.apache.iceberg.data.DeleteFilter#filter` not using this technique instead.

cc @rdblue

Reply:
I assume this has to do with Spark's partition at a time operation. In that case it can evaluate the filters in a streaming fashion.

See https://github.com/apache/iceberg/blob/c8b97c91ac04a2ee5ee8f746dcc4619a9c8d5ffe/data/src/main/java/org/apache/iceberg/data/DeleteFilter.java#L232

So for many filtered rows this might be slower in Spark. In Trino it's "disaster" in either case as it is doing Page at a time.
Although I will say that when the set of deleted rows is large one should try to formulate the delete as an equality delete anyway.

Note that the fix the Iceberg folks are proposing is to always memoize the filters.